source/unified-test-format/unified-test-format.rst

```diff
@@ -1405,20 +1405,18 @@ The structure of each object is as follows:
    corresponding language-specific component name. Test runners MUST assert
    that the actual component matches this value.
 
-- ``hasFailure``: Optional boolean. When this field is set to ``true``, 
-  the test runner MUST assert that the log message has a ``failure`` value
-  in its attached data. Then this field is set to ``false``, the test runner
-  MUST assert has no ``failure`` value in its attached data.
-  Currently, this assertion is only relevant for "command failed" log messages.
-  This value is often used in tandem with ``failureIsRedacted``.
-
-- ``failureIsRedacted``: Optional boolean. This field MUST only be specified
-  when ``hasFailure`` is present and its value is ``true``. When ``true``,
+- ``failureIsRedacted``: Optional boolean. This field SHOULD only be specified
```

Comment:
Nit: not sure how to interpret this combination of SHOULD and MUST.  Why aren't they both the same?

Reply:
I agree it feels funny; this comes back to the unified format's design rationale section on SHOULD vs MUST.

> This specification commonly uses "SHOULD" when providing guidance on writing test files. While this may appear contradictory to the driver mantra preferring "MUST", it is intentional. Some of this guidance addresses style (e.g. adding comments, using YAML anchors) and cannot be enforced with a JSON schema... The specification does prefer "MUST" in other contexts, such as discussing parts of the test file format that are enforceable by the JSON schema or the test runner implementation.

the "SHOULD" is used to describe something the schema allows but people shouldn't do (use `failureIsRedacted` without also asserting the failure `$$exists`) whereas the MUST describes something the test runner does.

Reply:
Gotcha